orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/AbstractServerGroupTask.groovy

```diff
@@ -60,7 +62,20 @@ abstract class AbstractServerGroupTask extends AbstractCloudProviderAwareTask im
     String account = getCredentials(stage)
 
     def operation = convert(stage)
-    validateClusterStatus(operation)
+
+    try {
+      validateClusterStatus(operation)
+    } catch (Exception e) {
+      def context = stage.getContext().findAll { it.value != null }
```

Comment:
Stage.context is a Map<String, Object> that may include null values but TaskResult.context is an ImmutableMap, so need to filter out null values.

Reply:
TBH not a huge fan, I don't believe we should be manipulating `stage.getContext()` directly here.

A more correct solution might some something along the lines of allowing the `TaskResult` to include a set of keys that should be removed from the context.

At this point, I would probably err on the side of simplicity and go with `RetrySupport` for 30s. I don't think that's the end of the world and it's something we're already doing in a few spots.

That said, I do think there is a _more_ generic opportunity to:

(a) track the # of times that a retryable task has been retried
(b) track the last exception encountered (and clear it if successful) such that it could be surfaced in the event of a timeout.

I would like both (a) and (b) to be something handled outside the scope of an individual task (ie. maybe the `RunTaskHandler` + an attribute of the message indicating how many times it's been retried).

The generic soln isn't something that needs to be handled within this PR.